File: Number_of_Lines_To_Write_String.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int lines=1;
        int width=0;
        
        for(char ch:s.toCharArray()){
            int value=widths[ch-'a'];
            width+=value;
            if(width>100){
                lines++;
                width=value;
            }
        }
      
        return new int[]{lines,width};
    }
}

File: Number_of_Lines_To_Write_String.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) 
    {
        int line = 1, currentLine = 0;
        
        for(int i=0; i<s.length(); i++)
        {
            if(currentLine + widths[s[i]-'a'] > 100)
            {
                line++;
                currentLine = widths[s[i]-'a'];
            }
            else
                currentLine += widths[s[i]-'a'];
        }
        
        vector<int> res;
        
        res.push_back(line);
        res.push_back(currentLine);
        
        return res;
    }
};
